go/vt/vtgateproxy/discovery.go

```diff
@@ -173,7 +173,11 @@ func (b *JSONGateResolverBuilder) start() error {
 			fileStat = checkFileStat
 
 			contentsChanged, err := b.parse()
-			if err != nil || !contentsChanged {
+			if err != nil {
+				log.Error(err)
+				continue
+			}
```

Comment:
we should be logging these errors, not sure if we care about incrementing `unchangedCount` in this case